source/opencascade/utilities.cc

```diff
@@ -267,6 +275,70 @@ namespace OpenCASCADE
     AssertThrow(OK, ExcMessage("Failed to write IGES file."));
   }
 
+
+  TopoDS_Shape
+  read_STL(const std::string &filename)
+  {
+    StlAPI_Reader reader;
+    TopoDS_Shape  shape;
+    reader.Read(shape, filename.c_str());
+    return shape;
+  }
+
+
+  void
+  write_STL(const TopoDS_Shape &shape,
+            const std::string & filename,
+            const double        deflection,
+            const bool          sew_different_faces,
+            const double        sewer_tolerance,
+            const bool          is_relative,
+            const double        angular_deflection,
+            const bool          in_parallel)
+  {
+    TopLoc_Location            Loc;
+    std::vector<TopoDS_Vertex> vertices;
+    std::vector<TopoDS_Edge>   edges;
+    std::vector<TopoDS_Face>   faces;
+    OpenCASCADE::extract_geometrical_shapes(shape, faces, edges, vertices);
+    bool mesh_is_present = true;
+    for (unsigned int i = 0; i < faces.size(); ++i)
+      {
+        Handle(Poly_Triangulation) theTriangulation =
+          BRep_Tool::Triangulation(faces[i], Loc);
+        if (theTriangulation.IsNull())
+          {
+            mesh_is_present = false;
+            break;
+          }
+      }
+    TopoDS_Shape shape_to_be_written = shape;
+    if (!mesh_is_present)
+      {
+        if (sew_different_faces)
+          {
+            BRepBuilderAPI_Sewing sewer(sewer_tolerance);
+            sewer.Add(shape_to_be_written);
+            sewer.Perform();
+            shape_to_be_written = sewer.SewedShape();
+          }
+        else
+          shape_to_be_written = shape;
+        BRepMesh_IncrementalMesh mesh_im(shape_to_be_written,
+                                         deflection,
+                                         is_relative,
+                                         angular_deflection,
+                                         in_parallel);
+      }
+
+    StlAPI_Writer writer;
+    const auto    error = writer.Write(shape_to_be_written, filename.c_str());
+#  if ((OCC_VERSION_MAJOR * 100 + OCC_VERSION_MINOR * 10) >= 690)
```

Comment:
@bangerth @luca-heltai  this is the other change, following the discussion of last week we should check the version and this way seems much more general than the first one I uploaded yesterday.